src/dropdown.ts

```diff
@@ -2304,35 +2301,17 @@ export default class Dropdown extends LitElement implements FormControl {
   }
 
   #onOptionsSelectedChange(event: Event) {
-    if (!this.multiple) {
-      for (const option of this.#optionElements) {
-        if (
-          option !== event.target &&
-          event.target instanceof DropdownOption &&
-          event.target.selected &&
-          option.selected &&
-          event.target !== this.#selectAllElementRef.value
-        ) {
-          option.selected = false;
-        }
-      }
+    if (this.#isSelectionFromSelectAllOrNone) {
```

Comment:
Rather than have guards all over the place below against Select All, we simply return now. `#selectAllOrNone()` now handles setting `#value` for itself.